### ISSUE TITLE
Issues compiling, LOG (FATAL) << control should never reach here

### ISSUE THREAD

**Issue by syndamn** [ceres-solver/ceres-solver#652]
When I install ceres through OpenMVG, it passes all the required tests. However, there appears to be some error on the ceres factor side, as I get the following error. I have checked and I have the same cost functions as the repo, I just dont understand how I would pass the proper syntax to get past these checks.  Thanks in advance for the help

**My current environment is:**
Ubuntu 18.04
gcc 7.5
ceres 1.13

**Error:**
Error using mex
In file included from /usr/local/include/ceres/cost_function_to_functor.h:94:0,
                 from /usr/local/include/ceres/ceres.h:40,
                 from
                 /home/sydneyid/satellite-pose-estimation/pose_refinement/SA-LMPE/ba/openmvg_ba.cpp:23:
                 /usr/local/include/ceres/dynamic_cost_function_to_functor.h: In constructor
                 ‘ceres::DynamicCostFunctionToFunctor::DynamicCostFunctionToFunctor(ceres::CostFunction*)’:
                 /usr/local/include/ceres/dynamic_cost_function_to_functor.h:108:5: error:
                 ‘CHECK_NOTNULL’ was not declared in this scope
     CHECK_NOTNULL(cost_function);
     ^~~~~~~~~~~~~
In file included from /usr/local/include/ceres/dynamic_numeric_diff_cost_function.h:44:0,
                 from /usr/local/include/ceres/ceres.h:44,
                 from
                 /home/sydneyid/satellite-pose-estimation/pose_refinement/SA-LMPE/ba/openmvg_ba.cpp:23:
                 /usr/local/include/ceres/internal/numeric_diff.h: In static member function ‘static
                 bool ceres::internal::NumericDiff<CostFunctor, kMethod, kNumResiduals, N0, N1, N2,
                 N3, N4, N5, N6, N7, N8, N9, kParameterBlock,
                 0>::EvaluateJacobianForParameterBlock(const CostFunctor*, const double*, const
                 ceres::NumericDiffOptions&, int, int, int, double**, double*)’:
/usr/local/include/ceres/internal/numeric_diff.h:438:9: error: ‘FATAL’ was not declared in this scope
     LOG(FATAL) << "Control should never reach here.";

**Comment by sandwichmaker** [comment]
This looks like a case where glog is not being included in some of the header files that refer to it.

**Comment by sandwichmaker** [comment]
I have just pushed a change which adds some missing glog/logging.h includes. Can you see if that fixes your problem?

**Comment by syndamn** [comment]
@sandwichmaker Thank you so much for your prompt response! That commit did fix my original error, but I am still having issues on the glog front.
I've verified that i do have libglog.a installed in /usr/local/lib. When I looked for solutions to this problem on stack overflow, they just suggested to disable glog, which isnt an option for what I want to do. Another source suggested adding glog to a cmake file, but I thought glog was already included in the ceres package? I apologize if this is a very basic question, I'm still learning how to script CMake files


My current error is (it essentially repeats these undefined reference errors for a long while)):
lock_sparse_matrix.cc:(.text+0x24d0): undefined reference to `google::LogMessageFatal::~LogMessageFatal()'
block_sparse_matrix.cc:(.text+0x2558): undefined reference to
`google::LogMessageFatal::LogMessageFatal(char const*, int, google::CheckOpString const&)'
block_sparse_matrix.cc:(.text+0x2560): undefined reference to `google::LogMessage::stream()'
/usr/local/lib/libceres.a(block_sparse_matrix.cc.o): In function
`ceres::internal::BlockSparseMatrix::CreateRandomMatrix(ceres::internal::BlockSparseMatrix::RandomMatrixOptions
const&)':
block_sparse_matrix.cc:(.text+0x26b0): undefined reference to `google::kLogSiteUninitialized'
block_sparse_matrix.cc:(.text+0x26d2): undefined reference to `google::LogMessage::LogMessage(char const*,
int)'
block_sparse_matrix.cc:(.text+0x26da): undefined reference to `google::LogMessage::stream()'
block_sparse_matrix.cc:(.text+0x26f6): undefined reference to `google::LogMessage::~LogMessage()'
block_sparse_matrix.cc:(.text+0x2ed3): undefined reference to `fLI::FLAGS_v'
block_sparse_matrix.cc:(.text+0x2eeb): undefined reference to `google::InitVLOG3__(int**, int*, char
const*, int)'
block_sparse_matrix.cc:(.text+0x2fd5): undefined reference to
`google::LogMessageFatal::LogMessageFatal(char const*, int, google::CheckOpString const&)'
block_sparse_matrix.cc:(.text+0x2fdd): undefined reference to `google::LogMessage::stream()'
block_sparse_matrix.cc:(.text+0x2fe5): undefined reference to `google::LogMessageFatal::~LogMessageFatal()'
block_sparse_matrix.cc:(.text+0x3030): undefined reference to
`google::LogMessageFatal::LogMessageFatal(char const*, int, google::CheckOpString const&)'
block_sparse_matrix.cc:(.text+0x3038): undefined reference to `google::LogMessage::stream()'
block_sparse_matrix.cc:(.text+0x308a): undefined reference to
`google::LogMessageFatal::LogMessageFatal(char const*, int, google::CheckOpString const&)'
block_sparse_matrix.cc:(.text+0x3092): undefined reference to `google::LogMessage::stream()'
block_sparse_matrix.cc:(.text+0x30e2): undefined reference to
`google::LogMessageFatal::LogMessageFatal(char const*, int, google::CheckOpString const&)'
block_sparse_matrix.cc:(.text+0x30ea): undefined reference to `google::LogMessage::stream()'
block_sparse_matrix.cc:(.text+0x313d): undefined reference to
`google::LogMessageFatal::LogMessageFatal(char const*, int, google::CheckOpString const&)'
block_sparse_matrix.cc:(.text+0x3145): undefined reference to `google::LogMessage::stream()'
block_sparse_matrix.cc:(.text+0x3191): undefined reference to
`google::LogMessageFatal::LogMessageFatal(char const*, int, google::CheckOpString const&)'
